### PR TITLE
feat: (teams): Make teams page static and remove teams state from redux

### PR DESCRIPTION
**Pull request by memoyil**
To review:

https://deploy-preview-3129--pancakeswap-dev.netlify.app/teams

### CODE REVIEW

File: src/pages/teams/[id].tsx [hunk 1]
```diff
@@ -1,3 +1,68 @@
...
+
+export const getStaticPaths: GetStaticPaths = () => {
+  return {
+    fallback: true,
+    paths: [],
+  }
+}
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  const teamIds = (params.id as string[]) || []
```
Comment (chef-jojo) [inline src/pages/teams/[id].tsx:30]: I'm curious how does it work, `params.id` should be string 🤯

File: src/views/Teams/index.tsx [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
 import TeamListCard from './components/TeamListCard'
 import TeamHeader from './components/TeamHeader'
+import { getTeams } from '../../state/teams/helpers'
+import { FetchStatus } from '../../config/constants/types'
 
 const Teams = () => {
   const { t } = useTranslation()
-  const { teams, isLoading } = useTeams()
-  const teamList = Object.values(teams)
+  const { data, status } = useSWRImmutable('teams', async () => getTeams())
```
Comment (chef-jojo) [inline src/views/Teams/index.tsx:14]: looks like if we use `useSWRImmutable` here, it won't trigger fetch on client

Comment (memoyil) [reply src/views/Teams/index.tsx]: Done 👍 changed to useSwr

File: src/pages/teams/[id].tsx (hunk 1 above)
Comment (memoyil) [reply src/pages/teams/[id].tsx]: Done

File: src/views/ProfileCreation/TeamSelection.tsx [hunk 3]
```diff
@@ -10,10 +12,19 @@ import useProfileCreation from './contexts/hook'
 const Team: React.FC = () => {
   const { teamId: currentTeamId, actions } = useProfileCreation()
   const { t } = useTranslation()
-  const { teams } = useTeams()
+  const [teams, setTeams] = useState<TeamsById>(teamsById)
```
Comment (chef-jojo) [inline src/views/ProfileCreation/TeamSelection.tsx:15]: why not useSWR here?

File: src/pages/teams/[id].tsx [hunk 4]
```diff
@@ -1,3 +1,68 @@
...
+
+  const isValidTeamId = teams.findIndex((team) => team.id === idNumber) !== -1
+  if (!isValidTeamId) {
+    return {
+      notFound: true,
+    }
+  }
+
+  const fetchedTeam = await getTeam(idNumber)
+  if (!fetchedTeam) {
```
Comment (chef-jojo) [inline src/pages/teams/[id].tsx:47]: Let's use try catch

File: src/pages/teams/index.tsx [hunk 5]
```diff
@@ -1,3 +1,43 @@
...
+      }}
+    >
+      <Teams />
+    </SWRConfig>
+  )
+}
+
+export const getStaticProps: GetStaticProps = async () => {
+  const fetchedTeams = await getTeams()
+  if (!fetchedTeams) {
```
Comment (chef-jojo) [inline src/pages/teams/index.tsx:22]: ditto

File: src/views/ProfileCreation/TeamSelection.tsx (hunk 3 above)
Comment (memoyil) [reply src/views/ProfileCreation/TeamSelection.tsx]: Fixed

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**